src/tensor_ops/matmul/cuda_kernel.rs

```diff
@@ -76,7 +453,66 @@ impl super::MatMatBrKernel<f32> for Cuda {
         grad_rhs: &mut Self::Storage<(Const<K>, N), f32>,
         grad_out: &Self::Storage<(B, M, N), f32>,
     ) -> Result<(), Self::Err> {
-        todo!()
+        let (batch, m, _) = lhs.shape;
+        let (k, n) = rhs.shape;
+        // TODO use strides
+        unsafe {
+            // grad_lhs += grad_out * rhs^T
+            let m_op = m.size() as i32;
+            let n_op = k.size() as i32;
+            let k_op = n.size() as i32;
+            let cfg = StridedBatchedConfig {
+                gemm: GemmConfig {
+                    transa: cublasOperation_t::CUBLAS_OP_T,
+                    transb: cublasOperation_t::CUBLAS_OP_N,
+                    m: n_op,
+                    n: m_op,
+                    k: k_op,
+                    alpha: 1.0,
+                    lda: k_op,
+                    ldb: k_op,
+                    beta: 1.0,
+                    ldc: n_op,
+                },
+                stride_a: 0,
+                stride_b: grad_out.strides[0] as i64,
+                stride_c: grad_lhs.strides[0] as i64,
+                batch_size: batch.size() as i32,
+            };
+            self.blas.gemm_strided_batched_async(
+                cfg,
+                rhs.data.as_ref(),
+                grad_out.data.as_ref(),
+                Arc::make_mut(&mut grad_lhs.data),
+            )?;
+        }
+        // TODO handle stride 0
+        let grad_rhs_buf = Arc::make_mut(&mut grad_rhs.data);
+        for b in 0..batch.size() {
```

Comment:
add note that these have to be sequential since cublas sgemm doesn't use atomicAdd